apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -13,42 +25,99 @@ type Properties = {
 };
 
 const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
-	const { control, errors, handleSubmit } = useAppForm<UserSignUpRequestDto>({
-		defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
-		validationSchema: userSignUpValidationSchema,
-	});
+	const [isPasswordHidden, setIsPasswordHidden] = useState<boolean>(true);
+	const [isConfirmPasswordHidden, setIsConfirmPasswordHidden] =
+		useState<boolean>(true);
+	const INPUT_ICON_SIZE = 20;
+
+	const handlePasswordIconPress = (): void => {
+		setIsPasswordHidden(!isPasswordHidden);
+	};
+
+	const handleConfirmPasswordIconPress = (): void => {
+		setIsConfirmPasswordHidden(!isConfirmPasswordHidden);
+	};
+
+	const { control, errors, handleSubmit, setError, watch } =
+		useAppForm<UserSignUpFormDto>({
+			defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
+			validationSchema: userSignUpValidationSchema,
+		});
 
 	const handleFormSubmit = useCallback((): void => {
-		void handleSubmit(onSubmit)();
-	}, [handleSubmit, onSubmit]);
+		void handleSubmit((data: UserSignUpFormDto) => {
+			const { password } = data;
+			const confirmPassword = watch(UserCustomValidation.FIELD);
```

Comment:
```suggestion
		void handleSubmit((data: UserSignUpFormDto) => {
			const { confirmPassword, ...userData } = data;
```
Use a better name instead of `data`